components/modal/core/Dialog.razor.cs

```diff
@@ -160,20 +160,12 @@ private void OnDialogMouseDown()
         }
 
         private async Task OnMaskMouseUp()
-        {
-            if (Config.MaskClosable && _dialogMouseDown)
-            {
-                await Task.Delay(4);
-                _dialogMouseDown = false;
-            }
-        }
-
-        private async Task OnMaskClick()
         {
             if (Config.MaskClosable && !_dialogMouseDown)
             {
                 await CloseAsync();
             }
+            _dialogMouseDown = false;
```

Comment:
I think it also need to add the  OnMaskMouseDown to set _dialogMouseDown=true, because some time the OnMaskMouseUp would not be trigger and then click on the mask, it doesn't close.

Reply:
The Dialog should only close when `OnDialogMouseDown = false`.
`_dialogMouseDown` will be set to true in `OnDialogMouseDown`, which is bind to `onmousedown` of the dialog window. When users mousedown in the dialog window (`_dialogMouseDown = true`) and then drag out and mouseup on the mask, it should not close the dialog. 
The element of dialog window is a child of mask, and there is no `stopPropagation` on the mouseup event. The `_dialogMouseDown` should be set to false correctly when user clicking inside the dialog window.

But you do remind me that if user add elements that have event `mousedown:stopPropagation` in the content, clicking on these elements will close the dialog incorrectly.

Reply:
I found that in antd (Node.Js), it has the same problem. Since setting `mousedown:stopPropagation` is not normal, maybe it's not a problem

Reply:
Please look at this screenshot, when you select the text and drag it to outside of modal, then click on the mask, the modal doesn't close until it was clicked again.

![modal-closing](https://github.com/user-attachments/assets/a963d4b5-e5ee-4f6d-a7b7-e5e8104633d5)

Reply:
I got what you mean. I should add `OnMaskMouseDown` to set `_dialogMouseDown = false` to prevent this problem. I will update the code tonignt.